site/content/blog/argocd-carvel-plugin.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Continuous delivery using a Carvel ytt Argo CD plugin"
+slug: argocd-carvel-plugin
+date: 2022-02-02
+author: Cari Lynn
+excerpt: "Use your favorite Carvel templating tool in your GitOps continuous delivery using the Carvel ytt Argo CD plugin."
+image: /img/logo.svg
+tags: ['ytt', 'argocd', 'gitops']
+---
+
+Argo CD is a declarative, GitOps, continuous delivery tool for kubernetes. It's design follows the GitOps philosophy of using Git as a single source of truth for the desired state of the system. In this example we're storing desired application state in ytt templates, and extending Argo CD to template and deploy them.
+
+At a high level a deployment using Argo CD starts with a configuration change. A commit with a change is made to the application repository, causing the Argo CD controller to notice the desired state has changed. It processes the manifests from the application repository through built-in templating engines like helm, or what we will be using: a Carvel ytt Argo CD plugin. Finally, it applies the manifests to the cluster via kubectl.
```

Comment:
> "to the cluster via kubectl"

i dont think it actually uses kubectl to apply it. i believe it has its own code that does the application of resources.

Reply:
I assumed so, however they mention it in the [docs](https://argo-cd.readthedocs.io/en/stable/getting_started/#7-sync-deploy-the-application)
> This command retrieves the manifests from the repository and performs a kubectl apply of the manifests.

But in the blog post this wording isn't necessary, I'm omit the kubectl mention.